group/assignment5/main.cpp

```diff
@@ -0,0 +1,96 @@
+#include "AlphabetFromDictionary.hpp"
+#include "codeu_test_lib.h"
+
+int main() {
+    vector<string> dictionary;
+    vector<char> alphabet;
+    bool result;
+    unordered_map<char, int> idxs;
+    vector<bool> expectedChars;
+    vector<bool> actualChars;
+
+    // test 1 - assignment example
```

Comment:
I still do believe that each test case should better be a separate function with a clear name. Maybe we should also print the whole alphabet in case one of the related tests fails... It's just for convenience, though.
In addition, if I replace "CAT" with empty string in this test case, I get a following error:
`main.cpp:30 Failed: Statement (idxs['R'] < idxs['C']) is false, expected true.`
It may be a bug as 'R' still must go before 'C', am I right?

Reply:
I like how tests in your assignment are organized, we can do the same thing here?

Reply:
Sure, if everyone agree, I can take care of that.

Reply:
The test fails because the algorithm checks only 2 adjacent words in the dictionary at a time, and really doesn't handle empty strings (my bad :( ). I thought about 2 solutions:
a.  Erase empty strings from the dictionary.
b. Save the last non-empty string in a variable, so we don't lose information when we encounter an empty string.

What do you think? Any other ideas?

Reply:
I think we can just skip them while looping through the dictionary, which is more or less equivalent to your b option.